guacamole/src/main/webapp/app/auth/service/authenticationService.js

```diff
@@ -394,5 +407,45 @@ angular.module('auth').factory('authenticationService', ['$injector',
 
     };
 
+    /**
+     * This function accepts a callback function and stores it in an array of 
+     * functions that will be run before page reload when a user logout is triggered.
```

Comment:
I recommend avoiding the temptation to document all functions as "This function ..." and instead write as if that has already been said. This is the general convention for documenting function behavior, and is the convention we follow elsewhere.

For example, instead of:

```js
/**
 * This function accepts a duration and beeps for that amount of time.
 *
 * @param {Number} duration
 *     The number of milliseconds that the beep should last.
 */
var beep = function beep(duration) {
   ...
};
```

Use:

```js
/**
 * Beeps for the specified amount of time.
 *
 * @param {Number} duration
 *     The number of milliseconds that the beep should last.
 */
var beep = function beep(duration) {
   ...
};
```